promql/engine.go

```diff
@@ -1363,17 +1364,28 @@ func (ev *evaluator) eval(expr parser.Expr) (parser.Value, annotations.Annotatio
 			}, e.Expr)
 		}
 
-		return ev.rangeEval(initSeries, func(v []parser.Value, sh [][]EvalSeriesHelper, enh *EvalNodeHelper) (Vector, annotations.Annotations) {
+		var warnings annotations.Annotations
+		if e.Op == parser.QUANTILE && e.Param != nil {
+			q, _ := strconv.ParseFloat(e.Param.String(), 64)
```

Comment:
This only works if `e.Param.String()` is directly representing a float number. However, it could be something like `1/2` or `3/2` . The latter should trigger the annotation, but it doesn't. Even worse, the parameter might even be the result of a TSDB lookup or otherwise depending on the evaluation time, e.g. `scalar(count(up))`. (This is a nonsensical example, of course, but maybe there is ever a valid use case for an evaluation-time dependent expression. And even if not, we have to treat this case correctly.).

The first example (where the parameter is step-invariant, but still a more complex PromQL expression) is quite valid, in any case.

So I do think we have to do the check after the engine has determined the value of `q`. Maybe there is a way to track if an annotations has already been added for this series, and then stop adding more for each step?